libraries/Wire/src/Wire.cpp

```diff
@@ -323,11 +331,10 @@ int TwoWire::read(void)
     value = rxBuffer[rxBufferIndex];
     ++rxBufferIndex;
 
-    /*  Reduce buffer size to free memory in case of large memory use when no more
-        data available */
-    if((rxBufferIndex == rxBufferLength) && (rxBufferLength > BUFFER_LENGTH)) {
-      rxBuffer = resetBuffer(rxBuffer);
-    }
+    /* Reset rx buffer when no more data available */
+    /*if(rxBufferIndex == rxBufferLength) {
+      resetRxBuffer();
+    }*/
   }
```

Comment:
Why keeping commented source code ?

Reply:
Just to show it is possible to reset the Rx buffer here. But think it is not required.